shopinvader_invoice/tests/common.py

```diff
@@ -74,6 +74,42 @@ def _check_data_content(self, data, invoices):
                 invoice.amount_total_signed,
             )
             self.assertEqual(current_data.get("amount_due"), invoice.amount_residual)
+
+            self.assertEqual(
+                len(current_data.get("lines")), len(invoice.invoice_line_ids)
+            )
+
+            # lines are not ordered (sorted)
+            # so this code checks each line of the invoice
+            # with each line of the output
+            number_of_lines_ok = 0
+            for current_line in current_data.get("lines"):
+                for line in invoice.invoice_line_ids:
+
+                    mappings = [
+                        # (in invoice line, in output)
+                        ("name", "name"),
+                        ("price_unit", "price"),
+                        ("price_subtotal", "untaxed"),
+                        ("price_total", "total"),
+                        ("quantity", "qty"),
+                    ]
+                    out = {}
+                    for mapping in mappings:
+                        out[mapping[1]] = line[mapping[0]]
+
+                    # sku is a bit special
+                    out["sku"] = line["product_id"]["default_code"]
```

Comment:
It's quite strange to do that. Why not `line.product_id.default_code`?

Reply:
to be consistent with previous lines (`line[mapping[0]]`)

but i'm ok with both syntax